lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1961,4 +1967,22 @@ protected getCachedTokenInternal(scopes : Array<string> , user: User): CacheResu
         return this._logger;
     }
 
+    /*
+    * Uses adal's id_token to achieve silent SSO.
+    */
+    private silentLogin(scopes: Array<string>, authority?: string, user?: User, extraQueryParameters?: string) {
+        this._silentLogin = true;
+        this.acquireTokenSilent(scopes, authority, user, extraQueryParameters)
+            .then((idToken) => {
+                this._silentLogin = false;
+                this._logger.info("Unified cache call is successful");
+                this._tokenReceivedCallback.call(this, null, idToken, null, Constants.idToken, this.getUserState(this._silentAuthenticationState));
+            }, (error) => {
+                this._silentLogin = false;
+                this._logger.error("Error occurred during unified cache ATS");
+                this.loginRedirectHelper(scopes, extraQueryParameters);
```

Comment:
you are always calling loginRedirectHelper in the error case but invoking this for both loginRedirect/loginPopup. This will break. We should use a flag to detect if it's sredirect or popup

Reply:
Fixed.